### PR TITLE
Moved network check into has required hardware

### LINKED ISSUE #1673: testUtils rejection when toggling Bluetooth or WiFi is not handled in our tests
The Promise.reject() that was introduced for toggling Bluetooth and WiFi in testUtils.js is not handled in our tests (UnitTest_app.js). As a result the tests are not started and device is not registered in coordination server which later aborts the tests.
You can see example in following test log (search for 'toggleBluetooth in one of android devices): [TestLog](https://github.com/ThaliTester/TestResults/tree/98710374837bb9b_Wifi_multicast_lock_evabishchevich/)

I think it should somehow pass the "reject" information further to coordination server so that this device may be disqualified from tests (similar like we do for native tests -  see global.nativeUTFailed flag). 

Anyway, we need to find way to turn bluetooth ON if it is turned off. Maybe we could use infrastructure for executing native tests and at the end of test executing toggle all radios ON.

### PR DESCRIPTION
**Pull request by andrew-aladev**
Closes #1673.

Client will be disqualified from test server if it has no ability to enable bluetooth.

---
This change is [Reviewable](https://reviewable.io/reviews/thaliproject/thali_cordovaplugin/1697)

### CODE REVIEW

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 1. task. (8fe1256)

**Comment by ThaliTester** [comment]
Test 99465247 (8fe1256) build started.

**Comment by ThaliTester** [comment]
Test (Success) 99465247 build is completed (8fe1256)

See https://github.com/ThaliTester/TestResults/tree/994652478fe1256_Moved_network_check_into_has_required_hardware_andrew-aladev/ for the logs

**Comment by ThaliTester** [comment]
Test 994652478fe1256(8fe1256) has failed

See https://github.com/ThaliTester/TestResults/tree/994652478fe1256_Moved_network_check_into_has_required_hardware_andrew-aladev/ for the fail logs

File: test/www/jxcore/lib/testUtils.js [hunk 1]
```diff
@@ -217,48 +219,72 @@ module.exports.tmpDirectory = tmpDirectory;
...
-          function (error, result) {
+  function checkNetworks() {
+    if (!ThaliMobile) {
+      ThaliMobile = require('thali/NextGeneration/thaliMobile');
+    }
+    return ThaliMobile.getNetworkStatus()
+      .then(function (networkStatus) {
+        var promises = [];
+        if (networkStatus.wifi === 'off') {
+          promises.push(toggleWifi(true));
```
Comment (chapko) [inline test/www/jxcore/lib/testUtils.js]: I don't like the idea that a function named `hasRequiredHardware` in fact changes radios state. I would rather see it as 2 seperate functions, for example `hasRequiredHardware` and `enableRequiredHardware`. It requires one extra line in runTests.js, but looks much cleaner

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 2. task. (7e4ad1f)

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 2. task. (950c14a)

**Comment by ThaliTester** [comment]
Test (Success) 99465247 build is completed (7e4ad1f)

See https://github.com/ThaliTester/TestResults/tree/994652477e4ad1f_Moved_network_check_into_has_required_hardware_andrew-aladev/ for the logs

**Comment by ThaliTester** [comment]
Test 99465247 (950c14a) build started.

**Comment by ThaliTester** [comment]
Test (Fail) 99465247 build is completed (950c14a)

See https://github.com/ThaliTester/TestResults/tree/99465247950c14a_Moved_network_check_into_has_required_hardware_andrew-aladev/ for the logs

**Comment by ThaliTester** [comment]
Test 994652477e4ad1f(7e4ad1f) has failed

See https://github.com/ThaliTester/TestResults/tree/994652477e4ad1f_Moved_network_check_into_has_required_hardware_andrew-aladev/ for the fail logs

**Comment by chapko** [comment]
:lgtm:

---

Reviewed 1 of 2 files at r1, 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[test/www/jxcore/runTests.js, line 125 at r3](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1697#-K_51k8yEXk0Y-R0UsLX:-K_51k8yEXk0Y-R0UsLY:bdm7b7u) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/950c14a12505bd4aca83e075ae3717cbeae3cd5b/test/www/jxcore/runTests.js#L125)):*
> ```JavaScript
>           .then(function (version) {
> 
>             return thaliTape.begin(
> ```

NB: that some deep nesting, but I don't see how it can be fixed without significant API changes and it is probably not worth it right now.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1697#-:-K_51hG8mr9l9oxVFWKe:bnfp4nl)*

**Comment by yaronyg** [comment]
:lgtm:

---

Reviewed 1 of 2 files at r1, 2 of 2 files at r3.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1697#-:-K_7-_sBSoUyUx_4ZdnH:bnfp4nl)*

**Pull request merged**